vertical-pod-autoscaler/e2e/full_vpa.go

```diff
@@ -58,26 +63,35 @@ var _ = fullVpaE2eDescribe("Pods under VPA", func() {
 		config, err := framework.LoadConfig()
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
-		vpaCRD := newVPA(f, "hamster-vpa", &metav1.LabelSelector{
+		vpaCRD = newVPA(f, "hamster-vpa", &metav1.LabelSelector{
 			MatchLabels: map[string]string{
 				"name": "hamster",
 			},
 		})
 
-		vpaClientSet := vpa_clientset.NewForConfigOrDie(config)
+		vpaClientSet = vpa_clientset.NewForConfigOrDie(config)
 		vpaClient := vpaClientSet.PocV1alpha1()
 		_, err = vpaClient.VerticalPodAutoscalers(ns).Create(vpaCRD)
 		gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
 	})
 
 	ginkgo.It("stabilize at minimum CPU if doing nothing", func() {
-		waitForSpecificCPURequestInPods(f, metav1.ListOptions{LabelSelector: "name=hamster"}, parseQuantityOrDie(minimalCPU))
+		waitForResourceRequestAboveThresholdInPods(f, metav1.ListOptions{LabelSelector: "name=hamster"}, apiv1.ResourceCPU, parseQuantityOrDie(minimalCPU))
 	})
 
 	ginkgo.It("have cpu requests growing with usage", func() {
 		rc.ConsumeCPU(600 * replicas)
-		waitForCPURequestAboveThresholdInPods(f, metav1.ListOptions{LabelSelector: "name=hamster"}, parseQuantityOrDie("500m"))
+		waitForResourceRequestAboveThresholdInPods(f, metav1.ListOptions{LabelSelector: "name=hamster"}, apiv1.ResourceCPU, parseQuantityOrDie("500m"))
+	})
+
+	ginkgo.It("have memory requests growing with OOMs", func() {
+		// Wait for any recommendation
+		waitForRecommendationPresent(vpaClientSet, vpaCRD)
+		// Restart pods to have limits populated
```

Comment:
Probably admission controller should set the limit even if there is no recommendation yet.